include/inviwo/core/io/serialization/deserializer.h

```diff
@@ -1145,14 +1132,25 @@ void Deserializer::deserialize(const std::string& key, T*& data) {
     }
 }
 
-template <class T, class D>
-void Deserializer::deserialize(const std::string& key, std::unique_ptr<T, D>& data) {
-    if (auto ptr = data.get()) {
-        deserialize(key, ptr);
-    } else {
-        deserialize(key, ptr);
-        data.reset(ptr);
+template <class T>
+void Deserializer::deserialize(const std::string& key, std::unique_ptr<T>& data) {
+    auto ptr = data.release();
+
+    if constexpr (util::HasGetClassIdentifier<T>::value) {
+        auto keyNode = retrieveChild(key);
+        if (!keyNode) return;
+
+        const std::string type_attr{
+            detail::getNodeAttribute(keyNode, SerializeConstants::TypeAttribute)};
+
+        if (ptr && !type_attr.empty() && type_attr != ptr->getClassIdentifier()) {
```

Comment:
This handles derived classes?

Reply:
so if you have a std::unique_ptr<Property> that currently is a BoolProperty but is was serialised as a StringProperty
then it will delete the BoolProperty and create a StringProperty

Reply:
Then one should have provided a converter instead?
I assume that this can cause some headaches later because one forgot (or was lazy) to do that

Reply:
this will atm not apply to our properties, since we don't deserialise them using a unique_ptr